File: solutions/csharp/cars-assemble/1/CarsAssemble.cs
using System;

internal static class AssemblyLine
{
	private const int _offSpeed = 0;
	private const int _maxSpeed = 10;
	private const int _carsPerHourPerLevel = 221;
	private const int _minutesPerHour = 60;

	public static double SuccessRate(int speed) =>
		speed switch
		{
			_offSpeed => 0,
			<= 4 => 1,
			<= 8 => 0.9,
			<= 9 => 0.8,
			<= _maxSpeed => 0.77,
			_ => throw new NotImplementedException($"Maximum speed is {_maxSpeed}."),
		};

	public static double ProductionRatePerHour(int speed) =>
		speed * _carsPerHourPerLevel * SuccessRate(speed);

	public static int WorkingItemsPerMinute(int speed) =>
		(int)(ProductionRatePerHour(speed) / _minutesPerHour);
}

File: solutions/java/cars-assemble/1/src/main/java/CarsAssemble.java
public class CarsAssemble {
	private static final int carsProducedPerHour = 221;
	private static final RangeMap<Integer, Double> successPercentageBySpeedRange
			= RangeMap.of(
				new Speed(0).to(0.0),
				new SpeedRange(1, 4).to(1.00),
				new SpeedRange(5, 8).to(0.90),
				new Speed(9).to(0.80),
				new Speed(10).to(0.77)
			);

    public double productionRatePerHour(int speed) {
	    return carsProducedPerHour * speed
			           * successPercentageBySpeedRange.get(speed);
    }

	public int workingItemsPerMinute(int speed) {
		return (int) (this.productionRatePerHour(speed) / 60);
	}
}
